lib/resty/healthcheck.lua

```diff
@@ -39,6 +39,47 @@ local re_find = ngx.re.find
 local bit = require("bit")
 local ngx_worker_exiting = ngx.worker.exiting
 local ssl = require("ngx.ssl")
+local escape_uri = ngx.escape_uri
```

Comment:
`ngx.escape_uri` might not be the proper function for this, as it will percent-encode and escape many values that are valid in an http header.

The more I think about this, maybe I was being hasty in recommending header sanitizing. Surprisingly (to me at least), a quick look around reveals that there are several examples of popular lua-resty-* code that takes header input but makes no attempt at sanitizing them:

* [lua-resty-http](https://github.com/ledgetech/lua-resty-http/blob/7552ac4a20d91240c03a6df111d94fc237c06d37/lib/resty/http.lua#L336-L350)
* [lua-resty-websocket](https://github.com/openresty/lua-resty-websocket/blob/34f6b41e538151e7a5f013ff9709c1708a9958fa/lib/resty/websocket/client.lua#L171-L175)

While I still _personally_ think we ought to strip `\r\n` from the header values, maybe someone will cite the above as a reason why we shouldn't.

So in the interest of not blowing up the scope on this PR, I think it might be better to punt on the sanitize task (remove the `ngx.escape_uri()` calls). I'd love to hear from someone else on the team if there's any more context to this topic though.

What do you think?

Reply:
We can leave the validation works to the users, just like Kong uses a custom validator for HTTP header keys.
https://github.com/Kong/kong/blob/08b5349be535d712d5a65415f4c08f02742385ba/kong/db/schema/typedefs.lua#L545-L554
IMO, this library should only accept pre-formatted values.